Obs sub id mag

stevechesley opened the pull request:
Resolving issues #32 and #35:

* `mag` is now restricted to range -5.0 to +35.0. 
* `obsSubID` is now `StringTypeW25`, allowing greater flexibility in observer assigned observation tags.
* Associated documentation updates.

federicaspoto reviewed (approved):
Looks good to me, @stevechesley. Thank you!

stevechesley commented:
@federicaspoto, I inadvertently clobbered the changes to the master XML from the first commit on this branch, and I neglected some documentation files. I think the branch is all in order now, and so you should look at this again.  This PR now addresses issues #32, #35 and #38, which I will close when I merge after your review.
